sig-and-wg/sig-core/proposals/asset-store-proposal.md

```diff
@@ -0,0 +1,274 @@
+    url: https://some.domain.com/main.js
+  bucketRef:
+    name: my-bucket
+```
+
+The optional information is the:
+- ConfigMap reference, that points to the ConfigMap that introduces [a new file](https://github.com/kyma-project/kyma/blob/master/resources/core/charts/service-catalog-addons/charts/instances-ui/templates/configmap.yaml) that is also sent to the bucket along with other files
+- Asset validation webhook reference to a service that performs the validation of fetched assets before they are uploaded to the bucket. The use cases are:
+  - validation of specific file against some specification
+  - security validation
+```
+apiVersion: assetstore.kyma-project.io/v1alpha1
+kind: Asset
+metadata:
+  name: my-direct-assets
+  namespace: default
+spec:
+  source:
+    direct: https://some.domain.com/my.json
+    validationWebhookService:
+        name: swagger-operations-svc
+        namespace: default
+        endpoint: "/validate"
+    rewrites:
+      - keyvalue: 
+          basePath: /test/v2
+  bucketRef:
+    name: my-bucket
+status:
+  phase: Ready # Failed/Pending
+  reason: Validation failure
+  message: "file is not valid against provided json schema"
+```
+- Asset mutation webhook reference to a service that acts similar to the validation service mentioned above, but mutates the asset instead of just validating. The use case is for example the asset rewriting through `regex` operation or `keyvalue` (like for example modification in json specification)
+
+
+## Asset Validation Webhook details
+
+Asset Store must provide a flexible way of validating the asset before they are uploaded to the bucket. Different use cases bring different validation requirements. The best way is to do it by introducing a webhook that will receive an asset for validation and reply to the controller with defined status schema
+
+1. Controller calls the `/validate` endpoint of the given service in a given namespace
+```
+validationWebhookService:
+  name: swagger-operations-svc
+  namespace: default
+  metadata:
+    pattern: \json|yaml
+  endpoint: "/validate"
+```
+2. The service gets the following payload and by default must reply within 1sec:
+```
+{
+    name: my-direct-assets
+    namespace: default
+    metadata:
+      pattern: \json|yaml
+    assets: {
+      name1: "content",
+      name2: "content"
+    }
+}
+```
+3. The controller:
+   - times out after 1sec because of no response and updates the status of the Asset to False
+   - gets response with successful or failing validation in the following for:
+```
+{
+   status: {
+     name1: {
+       status: Failure,
+       message: "much more details of the failure"
+     },
+     name2: {
+       status: Success,
+       message: "much more details"
+     }
+   }
+}
+```
+4. If at least one asset failed the validation the status of the Asset is set to False and proper message about a given asset is added to the status. Otherwise the status is set to True and asset uplodaded to the bucket.
```

Comment:
```suggestion
4. If at least one asset failed the validation, the status of the Asset is set to `False` and the proper message about a given asset is added to the status. Otherwise, the status is set to `True` and asset is uploaded to the bucket.
```